### PR TITLE
Add Knowledge for module responsibility and defensive scope

### LINKED ISSUE #54: Add Knowledge for module responsibility and defensive scope
## Problem

Modules are often grouped around a shared feature noun rather than one semantic responsibility and reason to change. This makes it easy for an integration module to absorb validation, compatibility, migration, trust, presentation, and hypothetical future policy merely because all of them concern the same domain object.

Defensive programming can amplify the mistake. An agent imagines an edge case, adds production behavior for it, adds a test, and then treats the test as evidence that the behavior belongs in the module. Each local addition appears reasonable while the module becomes less cohesive and begins to repeat upstream contracts or pre-implement downstream policy.

A recent Dragon Marketplace SDK adapter exposed this pattern. The intended module only needed to call typed SDK methods, decode unchecked responses, translate relevant failures, and interpret one unavoidable human-readable Source field. The initial implementation also reproduced the SDK accepted-input grammar, modeled hypothetical missing-method states, and added future trust and migration normalization rules. Those additions were removed after restating the module responsibility.

## Knowledge to capture

Create evergreen software-design Knowledge covering these principles:

- Define a module by the meaning it owns and one stable reason to change, not by a shared feature or domain noun.
- Distinguish a module interface, its implementation, the seam where the interface lives, and an adapter that satisfies the interface.
- Measure useful depth by caller leverage and hidden complexity. Implementation complexity that adds caller concepts without leverage is fake depth.
- At an external integration seam, keep the adapter to the translation callers currently need: invoke the external interface, decode unchecked output, and translate relevant failures.
- Leave an upstream dependency accepted-input language with that dependency. Leave product, migration, trust, presentation, and other domain policy with their semantic owners.
- Admit defensive behavior only when a realistic value accepted by the interface can reach an interpretation or side effect the module owns. Name the producer, failure, and owner before adding the branch.
- A current approved expand-stage task may establish scope before production consumers switch. A merely possible future consumer does not.
- Tests protect behavior already owned by the module; an imagined test case cannot grant the module a new production responsibility.

## Decision procedure

The Knowledge should give authors and reviewers a compact way to assess each proposed branch, type, export, validation, and test:

1. What current requirement or concrete consumer needs it?
2. Which module is the semantic owner of the decision?
3. What event or decision would make this code change?
4. Does the external dependency already own the behavior?
5. Can the state occur through the real interface and execution path?
6. If this module were deleted, would the complexity disappear or spread to its callers?

Logic that changes for a different reason should move to its owner or remain deferred.

## Examples and boundaries

Include representative contrasting examples:

- Reject speculative trust or migration policy added to an SDK adapter for a future consumer.
- Keep validation at a filesystem or privileged side-effect boundary when the module genuinely interprets and acts on the admitted value.
- Allow an explicitly approved expand-stage module even when consumer cutover is a later ticket.
- Distinguish a legitimately thin external adapter from the deeper domain module that consumes it; do not turn the guidance into a rule that every module or every adapter must have a particular size.

## Placement and routing

- Add an independently retrievable Knowledge leaf, with the repository-required Scope and When to update sections, under the most appropriate software-design or programming category.
- Add it to the Knowledge index.
- Evaluate which code-authoring, code-review, test-authoring, and test-review Skills should route to it. Link to one authoritative Knowledge source rather than duplicating the principles across Skills.
- Keep the Dragon Marketplace incident as authoring evidence or a small illustrative example, not as project-specific Knowledge.

## Acceptance criteria

- The Knowledge separates responsibility-based module design from topic-based file organization.
- It explains semantic ownership, reasons to change, module depth, seams, adapters, speculative scope, and defensive-scope admission as one coherent model.
- It distinguishes runtime validation owned by a module from repeating an upstream contract.
- It states that tests follow production ownership rather than creating it.
- It includes the three boundary examples above and avoids line-count or universal thin-adapter rules.
- The Knowledge has one authoritative home, is indexed, and relevant Skills route to it without semantic duplication.
- Repository Knowledge, formatting, and link validation pass.

### PR DESCRIPTION
**Pull request by Soulike**
Closes #54

## Summary

- add evergreen software-design Knowledge for semantic module ownership, interface depth, external adapters, defensive-scope admission, staged scope, and test ownership
- index the Knowledge and add narrow conditional routes from library adoption, review-feedback handling, test authoring, and test review
- bump the primary plugin version to `2026.8.31-2`

## Design and behavioral evidence

The task is to help authors and reviewers decide whether a proposed branch, validation, policy, export, adapter behavior, or test belongs in a module. The Knowledge is independently retrievable from a module or integration-design task; the Skills only select it when their workflow encounters that responsibility decision.

Research and evidence included the Dragon Marketplace adapter case from #54, the existing Test effectiveness ownership model, Parnas on decomposition by hidden design decisions, Ousterhout on module depth, Cockburn on ports and adapters, and Fowler on deferring speculative capability. The project-specific incident remains authoring evidence; the maintained Knowledge is downstream-project-independent.

The new leaf owns module responsibility and defensive-scope admission. Test oracle, layer, and double decisions remain in Test effectiveness; security control mechanics remain in the security references; upstream accepted-input contracts remain with their dependencies. One leaf was retained because defensive admission, speculative scope, and test ownership are applications of the same semantic-ownership decision and share one reading and maintenance lifecycle.

Independent forward tests covered: a typed SDK adapter with speculative trust and migration policy; review feedback proposing hypothetical missing-method handling; a test request for an interface-impossible state; a suite whose setup reveals mixed module responsibilities; and ordinary test work for an established reachable behavior. Exact-head AI review exposed three selector-timing gaps; commit `26d8586` moved adapter selection to the post-package decision, mapped complete feedback before selecting Knowledge, and made test ownership depend on independent requirements and reachability rather than existing speculative code. A second exact-head review found that coverage dispositions could still cement unowned behavior; commit `bb993f2` added a production-responsibility gate and explicit disposition before coverage selection. Follow-up positive and negative scenarios verified both remediations without loading the Knowledge for ordinary direct SDK use or established presentation and compatibility tests. An independent base-to-head semantic comparison found no missing requirement, duplicate authority, route gap, or downstream-project dependency.

## Validation

- `python3 /Users/soulike/.codex/skills/.system/skill-creator/scripts/quick_validate.py` for each changed Skill
- `pnpm check`
- `git diff --check origin/main...HEAD`
- `node .github/scripts/plugin-version/check.ts origin/main HEAD`

### CODE REVIEW

File: skills/evaluate-library-adoption/SKILL.md [hunk 1]
```diff
@@ -93,8 +101,8 @@ After selecting a suitable installed package or reaching an explicit choice,
 return the approach and its evidence to the original task:
 
 - When an established solution is selected, use its supported interface and
-  keep project-specific policy outside it. Add an adapter only when it creates
-  a meaningful project boundary.
+  apply the loaded module-responsibility guidance when an adapter or surrounding
```
[the remark this answers is not in the document]
Comment (Soulike) [reply skills/evaluate-library-adoption/SKILL.md]: Fixed in 26d8586. The Knowledge selector now runs after the solution is selected and covers using or proposing an adapter, deciding whether one is justified, and placing surrounding policy. This restores the original meaningful-boundary decision while still avoiding the extra load for direct SDK use with no adapter or policy question. Verified with focused forward tests, quick_validate.py, and the full pnpm check.

File: skills/handle-code-review-feedback/SKILL.md [hunk 2]
```diff
@@ -30,17 +30,24 @@ Keep these disciplines throughout the workflow:
...
+   than as governing instructions.
 2. Identify the revision and code state to which the feedback applies. Check
    whether its concern still applies to the current work; an outdated or
    resolved marker does not prove that the underlying concern was addressed.
 3. Gather enough context to evaluate the actual claim. Inspect the relevant
    code paths, behavior, requirements, tests, history, platform constraints,
    or other sources as the concern requires; this is not an exhaustive input
    checklist.
-4. Read the complete supplied feedback set and available thread context before
+4. Resolve paths relative to this `SKILL.md`, then read
```
[the remark this answers is not in the document]
Comment (Soulike) [reply skills/handle-code-review-feedback/SKILL.md]: Fixed in 26d8586. The workflow now reads and maps the complete feedback set and available thread context before applying the module-responsibility selector, so concerns discovered only from fetched review threads cannot skip the Knowledge. Verified with a thread-only feedback scenario, quick_validate.py, and the full pnpm check.

File: skills/write-effective-tests/SKILL.md [hunk 3]
```diff
@@ -12,24 +12,29 @@ description: Decide whether a defined production behavior needs automated covera
...
    [Test effectiveness](../../knowledge/software-testing/test-effectiveness.md).
-3. Read
+3. Define the target production behavior or bounded coverage need. Discover its
+   owning code, existing tests and static gates, project-declared test locations
+   and commands, and local testing conventions. When the request concerns an
+   actual production change, inventory every behaviorally relevant change so no
+   required protection disappears behind the proposed tests.
+4. Read
+   [Module responsibility and defensive scope](../../knowledge/software-design/module-responsibility-and-defensive-scope.md)
+   when the request, available production contract, and discovered code do not
```
[the remark this answers is not in the document]
Comment (Soulike) [reply skills/write-effective-tests/SKILL.md]: Fixed in 26d8586. The selector no longer treats discovered source code as ownership evidence: it loads when a test request is used to establish production responsibility or when the accepted requirement, supported path, or semantic owner is not independently established. Existing speculative code therefore cannot authorize its cementing test, while established presentation and compatibility tests avoid unnecessary loading. Verified with focused positive and negative scenarios, quick_validate.py, and the full pnpm check.

File: skills/write-effective-tests/SKILL.md [hunk 4]
```diff
@@ -12,24 +12,30 @@ description: Decide whether a defined production behavior needs automated covera
...
+   owning code, existing tests and static gates, project-declared test locations
+   and commands, and local testing conventions. When the request concerns an
+   actual production change, inventory every behaviorally relevant change so no
+   required protection disappears behind the proposed tests.
+4. Read
+   [Module responsibility and defensive scope](../../knowledge/software-design/module-responsibility-and-defensive-scope.md)
+   when the test request is being used as evidence that production should own
+   defensive, compatibility, migration, trust, presentation, or future-consumer
+   behavior, or when the accepted requirement, supported execution path, or
+   semantic owner is not independently established.
```
[the remark this answers is not in the document]
Comment (Soulike) [reply skills/write-effective-tests/SKILL.md:25]: Fixed in bb993f2. The Skill now applies the loaded module-responsibility guidance before every coverage disposition, independently establishes the accepted requirement, supported path, and semantic owner, and assigns Production responsibility not established when that evidence is missing. That branch cannot design a cementing test or edit production without separate surrounding-task authority; only established production behavior reaches the existing coverage dispositions. Verified with speculative-code, established defensive behavior, external-validation, and ambiguous-owner scenarios, plus quick_validate.py and the full pnpm check.
